p2p/net/swarm/swarm_dial.go

```diff
@@ -382,6 +393,11 @@ func (s *Swarm) resolveAddrs(ctx context.Context, pi peer.AddrInfo) ([]ma.Multia
 			}
 		}
 
+		if len(toResolve) >= maximumToResolveAddresses {
+			// No point in resolving any more addresses, we don't have space.
+			continue
+		}
```

Comment:
Should this check be before the previous block that does resolution with Transport? 

If we don't have space in resolve list, there's not much the transport resolution can do either.